### PR TITLE
Automatically populate `apiURL` with javadoc.io

### PR DESCRIPTION
**Pull request by armanbilge**
Only if there's not already something set.

### CODE REVIEW

File: sonatype/src/main/scala/org/typelevel/sbt/TypelevelSonatypePlugin.scala [hunk 1]
```diff
@@ -35,17 +35,19 @@ object TypelevelSonatypePlugin extends AutoPlugin {
...
-          List(
-            "reload",
-            "project /",
-            "+mimaReportBinaryIssues",
-            "+publish",
-            "tlSonatypeBundleReleaseIfRelevant"))
-      )
+    Seq(
+      tlSonatypeUseLegacyHost := true,
+      autoAPIMappings := true
```
Comment (armanbilge) [inline sonatype/src/main/scala/org/typelevel/sbt/TypelevelSonatypePlugin.scala:40]: There's no reason this is controversial, right?

Comment (rossabaker) [reply sonatype/src/main/scala/org/typelevel/sbt/TypelevelSonatypePlugin.scala:40]: "This requires that dependencies have this information in its metadata" -- I don't know the consequences if they don't.  It just doesn't link, right?

**Review by rossabaker — APPROVED** [review]
I like it.

File: sonatype/src/main/scala/org/typelevel/sbt/TypelevelSonatypePlugin.scala (hunk 1 above)
Comment (armanbilge) [reply sonatype/src/main/scala/org/typelevel/sbt/TypelevelSonatypePlugin.scala:40]: Yep, I assume so.

**Pull request merged**